src/dxtbx/format/FormatNXmxEigerFilewriter.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import re
+
+import h5py
+
+from scitbx.array_family import flex
+
+import dxtbx.nexus
+from dxtbx.format.FormatNXmx import FormatNXmx
+from dxtbx.nexus import _dataset_as_flex, get_detector_module_slices
+
+DATA_NNNNNN_RE = re.compile(r"data_\d{6}")
+
+
+class FormatNXmxEigerFilewriter(FormatNXmx):
+
+    _cached_file_handle = None
+
+    @staticmethod
+    def understand(image_file):
+        with h5py.File(image_file) as handle:
+            if "/entry/instrument/detector/detectorSpecific/eiger_fw_version" in handle:
+                return True
+        return False
```

Comment:
We should probably fake up a test set for this? But this is not critical for merging. 

Should maek an issue to add modern Eiger from file writer data to test suite